path_login.go

```diff
@@ -188,6 +188,9 @@ func (b *kubeAuthBackend) aliasLookahead(ctx context.Context, req *logical.Reque
 		return logical.ErrorResponse(fmt.Sprintf("invalid role name %q", roleName)), nil
 	}
 
+	b.l.RLock()
+	defer b.l.RUnlock()
+
```

Comment:
I wonder if it would be safer to move this to line 183, so it's above the call to `b.role()`. Just noticing that is where the lock acquisition happens in `pathLogin()`:

https://github.com/hashicorp/vault-plugin-auth-kubernetes/blob/8aea07aa022cdc782d88ef1f189e4cd5f53146a8/path_login.go#L68-L74

Reply:
Fixed!